src/webgpu/shader/execution/expression/binary/af_data.ts

```diff
@@ -269,28 +524,15 @@ export function getAdditionAFInterval(lhs: number, rhs: number): FPInterval {
   return kAdditionTable[hashRawDataInputs(lhs, rhs)];
 }
 
-/** Simplified version of kSparseVectorF64Range that only uses values from sparseScalarF64Range() */
-export const kSparseVectorAFValues = {
-  2: sparseScalarF64Range().map(f => Array(2).fill(f)),
-  3: sparseScalarF64Range().map(f => Array(3).fill(f)),
-  4: sparseScalarF64Range().map(f => Array(4).fill(f)),
-} as const;
+/** Table mapping subtraction binary inputs to expectation intervals */
+const kSubtractionTable: Readonly<Record<number, FPInterval>> = Object.fromEntries(
+  kSubtractionRawValues.map(value => [
+    hashRawDataInputs(value.lhs, value.rhs),
```

Comment:
This assumes a perfect hash, but I don't see anything that would detect and error on collisions. That seems... hairy.

Reply:
Done